packages/rocket-uploads3-store-event-aws-infrastructure/src/s3-notification-event-store.ts

```diff
@@ -0,0 +1,68 @@
+import { Duration, RemovalPolicy, Stack } from '@aws-cdk/core'
+import { Bucket, EventType } from '@aws-cdk/aws-s3'
+import { RocketUtils } from '@boostercloud/framework-provider-aws-infrastructure'
+import { Function, Runtime, Code } from '@aws-cdk/aws-lambda'
+import { Table } from '@aws-cdk/aws-dynamodb'
+import { S3EventSource } from '@aws-cdk/aws-lambda-event-sources'
+import { createPolicyStatement } from '@boostercloud/framework-provider-aws-infrastructure/dist/infrastructure/stacks/policies'
+
+const path = require('path')
+
+export type AWSS3EventsLambdaParams = {
+  bucketName: string
+  eventTypeName: string
+  entityTypeName: string
+}
+
+export class S3NotificationEventStoreStack {
+  public static mountStack(params: AWSS3EventsLambdaParams, stack: Stack): void {
+    const appName = stack.stackName.split('-app')[0]
+
+    // Create new bucket to upload file
+    const sourceBucket = new Bucket(stack, 'sourceUploadBucket' + params.entityTypeName, {
+      bucketName: params.bucketName,
+      removalPolicy: RemovalPolicy.DESTROY,
+    })
```

Comment:
I guess this is more a matter of styles, but I personally like comments better and I think they abstract the idea of your code without looking the code. 

In any case, I think we should define a coding style for CDK resources for upcoming PRs and maybe now is the time to discuss about this hehe.

Any thoughts? @charlietfe @juanjoman @javiertoledo @alvaroloes @NickSeagull ?

Reply:
But then a function name is the same as adding the comment, plus, the fact that it is a machine (compiler/interpreter) delimited way of grouping code of the same responsibility. For the same reason, while code evolves, comments are much harder to maintain. Ending up in stuff like this 😅

![not basmati rice](https://pbs.twimg.com/media/DaTX9R1VQAEWWIK?format=jpg&name=900x900)

I believe that code comments are useful for explaining design decisions, e.g. in this same PR the decision of not passing the `eventStore` as a parameter, but not for anything else (of course, there are exceptions, but very minimal).

I think that we should avoid comments here as well, code is still code, be it CDK or not. If YAML had functions, I'd probably use them there 😂

Reply:
hahaha I agree with you @NickSeagull 

I've mainly decided to use comments in this specific case for 2 reasons

- Since we have infra code I would expect this not too change (or very little) compared to business logic code
- CDK documentation is really crap in most cases so having comments would help people explore te code 

If you don't mind I will post this in Slack and I'd like to know what everyone thinks hehe.